lib/ipam/ipam.go

```diff
@@ -299,10 +301,44 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 		logCtx = logCtx.WithField("handle", *handleID)
 	}
 	logCtx.Info("Looking up existing affinities for host")
-	affBlocks, err := c.blockReaderWriter.getAffineBlocks(ctx, host, version, pools)
+	affBlocks, affBlocksToRelease, err := c.blockReaderWriter.getAffineBlocks(ctx, host, version, pools)
 	if err != nil {
 		return nil, err
 	}
+
+	// Release any blocks still affine to this host but no longer selected by the larger ip pool.
```

Comment:
I think we should also mention that we only do this for blocks which have no IPs allocated, since that's an important qualification here.

Reply:
Also, "selected by the larger IP pool" is probably more accurately stated as "no longer part of an IP pool which selects this node".